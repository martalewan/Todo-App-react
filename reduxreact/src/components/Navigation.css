.navigation {
    position: sticky;
    left: 0;
    top: 0;
    z-index: 1;
    background-color: var(--bottle);
  }
  
  .logo {
    color: var(--creemy);
    padding-left: 5%;
    padding-right: 5%;
    padding-top: 1rem;
    padding-bottom: 1rem;
    font-family: 'Antic Didone', serif;
    font-weight: 800;
    font-size: 35px;
    letter-spacing: 3px;
  }
  