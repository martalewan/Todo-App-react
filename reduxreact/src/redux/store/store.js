import { createStore } from 'redux';
import rootReducer from '../reducers/todos';

const persistedState = localStorage.getItem('todos')
  ? JSON.parse(localStorage.getItem('todos'))
  : {};

const store = createStore(rootReducer,
  persistedState,
  /* eslint-disable no-underscore-dangle */
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__());

store.subscribe(() => {
  localStorage.setItem('todos', JSON.stringify(store.getState()));
});

export default store;
