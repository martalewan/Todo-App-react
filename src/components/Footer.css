.footer-wrapper {
    position: absolute;
    left: 0;
    bottom: 0;
    height: 50px;
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: var(--creemy);
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.095);
}

.footer-wrapper__link {
    margin: 0 2rem;
    margin-left: 5%;
    margin-right: 5%;
    font-size: 12px;
    color: var(--bottle);
}

.footer-wrapper__link a {
    color: var(--bottle);
}