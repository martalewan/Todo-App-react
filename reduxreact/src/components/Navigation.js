import './Navigation.css';

const Navigation = () => (
  <nav className="navigation">
    <p className="logo">ASPEN</p>
  </nav>
);

export default Navigation;
