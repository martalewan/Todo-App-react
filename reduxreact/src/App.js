import React from 'react';
import './App.css';
import Main from './components/Main';
import Navigation from './components/Navigation';
import Footer from './components/Footer';

const App = () => (
  <section className="App">
    <Navigation />
    <Main />
    <Footer />
  </section>
);

export default App;
