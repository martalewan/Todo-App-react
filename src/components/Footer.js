import './Footer.css';

const Footer = () => (
    <footer className="footer-wrapper">
        <p className="footer-wrapper__link">Made by <a href="https://martalewan.com/">Marta Lewan</a></p>
        <p className="footer-wrapper__link">2021</p>
    </footer>
);

export default Footer;
