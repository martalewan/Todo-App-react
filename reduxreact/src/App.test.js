import React from 'react'; 
import { render } from '@testing-library/react';
import App from './App';
import ReactDOM from 'react-dom';
import Navigation from './components/Navigation';
import Footer from './components/Footer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import store from './redux/store/store';

test('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(
    <Provider store={store}>
      <App />
    </Provider>, div
  );

  // ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('Snapshot', () => {
  const { asFragment } = render(<Navigation />)
  expect(asFragment()).toMatchSnapshot();
});

test('Snapshot', () => {
  const { asFragment } = render(<Footer />)
  expect(asFragment()).toMatchSnapshot();
});

