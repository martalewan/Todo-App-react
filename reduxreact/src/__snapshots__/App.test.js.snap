// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navigation"
  >
    <p
      class="logo"
    >
      ASPEN
    </p>
  </nav>
</DocumentFragment>
`;

exports[`Snapshot 2`] = `
<DocumentFragment>
  <footer
    class="footer-wrapper"
  >
    <p
      class="footer-wrapper__link"
    >
      Made by 
      <a
        href="https://martalewan.com/"
      >
        Marta Lewan
      </a>
    </p>
    <p
      class="footer-wrapper__link"
    >
      2021
    </p>
  </footer>
</DocumentFragment>
`;
